botocore/regions.py

```diff
@@ -510,6 +513,19 @@ def _validate_account_id_endpoint_mode(self):
             )
             raise InvalidConfigError(error_msg=error_msg)
 
+    def resolve_credential_scope_builtin(
+        self, builtin_configured, builtin_value
+    ):
+        """Resolve the ``AWS::Auth::CredentialScope`` builtin."""
+        if not builtin_configured or self._credentials is None:
+            return None
+
+        if builtin_value is not None:
+            return builtin_value
+
+        frozen_creds = self._credentials.get_frozen_credentials()
```

Comment:
Is this introducing a risk for mismatches in values? We call `get_frozen_credentials` separately in `resolve_account_id_builtin` and `resolve_credential_scope_builtin` which may both [invoke `_refresh`](https://github.com/boto/botocore/blob/4181e681982e60cf49285e17d21c35737299d78c/botocore/credentials.py#L611) at different times.

I think this got called out in the other PR at one point, but how are you ensuring the values don't change between the time each of these is resolved? Shouldn't they all be using the same set of frozen credentials? What testing did we do to ensure this was safe?

Reply:
I called this out in the design doc. Neither account ID nor credential scope are expected to change throughout the lifetime of a client so I opted against pulling frozen creds out to a single call. We can add a frozen credentials property here though.

If there were a risk however, using the same frozen credentials instance wouldn't solve the underlying race condition problem because there is still a separate call at the outset of request signing.